src/iosource/PktSrc.h

```diff
@@ -98,11 +98,23 @@ class PktSrc : public IOSource
 	 */
 	const char* ErrorMsg() const;
 
+	/**
+	 * Returns a reason why BPF compilation failed. This can be called
+	 * if PrecompileBPFFilter or PrecompileFilter returns false to
+	 * determine a reason.
+	 *
+	 * @return a pair containing a bool indicating whether the error is
+	 * considered catastrophic and an error string from the compilation.
+	 * If the bool is true, this indicates that setup using the filter
+	 * should exit with an error.
+	 */
+	std::pair<bool, std::string> FilterCompilationError() const { return filter_error; }
```

Comment:
> Maybe GetFilerCompilationError() ?

We have lots and lots of places in the code base where there is no `Get` on methods that simply return a value. I don't know why it's like that, but this just follows that.

> Also wonder if a struct with .fatal or .catastrophic (or even enum status code OK, FATAL, ... rather than bool) would be worth it?

That sounds like a good idea. `bool` is never terribly descriptive.